### ISSUE TITLE
[a11y][Android][regression] Checkbox semantics doesn't report "not checked" steady state on Android

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#191818]
### Steps to reproduce

1. Create an app that uses a Checkbox widget or just `Semantics(checked: isChecked)`.
2. Run on Android and enable TalkBack.
3. Touch the checkbox to focus it with TalkBack

### Expected results

TalkBack should read "checked" if the checkbox is checked and "not checked" if not.

### Actual results

Reads "checked" but does not read "not checked".

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  bool _isChecked = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: CheckboxListTile(
          title: Text('Toggle me!'),
          value: _isChecked,
          onChanged: (value) {
            if (value == null) return;
            setState(() {
              _isChecked = value;
            });
          },
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/e35255ca-df0e-45ea-850f-7dc947495418

</details>

### Logs

<details open><summary>Logs</summary>

N/A, google internal client

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

N/A, google internal client

</details>

**Comment by math1man** [comment]
Internal b/508138787, client huggsy

**Comment by saurabh-mirajkar** [comment]
@math1man Thanks for reporting this. I can confirm this behaviour on both stable & master channels using a Pixel 6 (Android 17). Routing this issue to the team for further investigation.

<details open><summary>stable</summary>

```console
Flutter 3.47.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 6655482ec0 (8 days ago) • 2026-08-19 10:07:23 -0700
Engine • hash 11d79658c444477b06513d32b52c8c4ccb7276b0 (revision 5d53178869) (8 days ago) • 2026-08-18 23:36:01.000Z
Tools • Dart 3.13.1 • DevTools 2.60.0
``` 
</details>

<details open><summary>master</summary>

```console
Flutter 3.48.0-1.0.pre-458 • channel master • https://github.com/flutter/flutter.git
Framework • revision fb86918005 (8 hours ago) • 2026-08-26 23:34:29 +0000
Engine • hash 41ed0cec7af8900580f0d1f742175c272203d48e (revision 73611d6c13) (2 hours ago) • 2026-08-26 21:56:46.000Z
Tools • Dart 3.14.0 (build 3.14.0-170.0.dev) • DevTools 2.61.0-dev.0
``` 
</details>

**Comment by jesswrd** [comment]
@saurabh-mirajkar can you bisect the Flutter version where this failure first appeared?

**Comment by math1man** [comment]
If it helps, our QA team first reported the issue on April 30. Sorry it took so long to file here.

**Comment by hannah-hyj** [comment]
From triage: Maybe we need to map the hasCheckableState flag or check box role to a role/class on android side

**Comment by saurabh-mirajkar** [comment]
> [@saurabh-mirajkar](https://github.com/saurabh-mirajkar) can you bisect the Flutter version where this failure first appeared?

Sorry for the delayed follow-up, @jesswrd . I’ve checked several stable releases (3.35.7, 3.38.10, 3.41.0, and 3.47.2), and the behavior is present in all of them.

> If it helps, our QA team first reported the issue on April 30. Sorry it took so long to file here.

 @math1man , could you please check with your QA team for the last Flutter version where TalkBack correctly announced the unchecked state ("not checked")? That would help us narrow down the regression.

**Comment by math1man** [comment]
We are a google3 app, so we are built from ~head version of Flutter. The release where our QA reported the issue was cut at 903189407, on April 21. That's all we know.
